analyzer/analyzer_test.go

```diff
@@ -0,0 +1,117 @@
+package analyzer
+
+import (
+	"context"
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/aquasecurity/fanal/extractor"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockDockerExtractor struct {
+	saveLocalImage  func(ctx context.Context, imageName string) (io.Reader, error)
+	extractFromFile func(ctx context.Context, r io.Reader, filenames []string) (extractor.FileMap, error)
+	extract         func(ctx context.Context, imageName string, filenames []string) (extractor.FileMap, error)
+}
+
+func (mde mockDockerExtractor) Extract(ctx context.Context, imageName string, filenames []string) (extractor.FileMap, error) {
+	if mde.extract != nil {
+		return mde.extract(ctx, imageName, filenames)
+	}
+	return extractor.FileMap{}, nil
+}
+
+func (mde mockDockerExtractor) ExtractFromFile(ctx context.Context, r io.Reader, filenames []string) (extractor.FileMap, error) {
+	if mde.extractFromFile != nil {
+		return mde.extractFromFile(ctx, r, filenames)
+	}
+	return extractor.FileMap{}, nil
+}
+
+func (mde mockDockerExtractor) SaveLocalImage(ctx context.Context, imageName string) (io.Reader, error) {
+	if mde.saveLocalImage != nil {
+		return mde.saveLocalImage(ctx, imageName)
+	}
+	return nil, nil
+}
+
+func (mde mockDockerExtractor) ExtractFiles(layer io.Reader, filenames []string) (extractor.FileMap, extractor.OPQDirs, error) {
+	panic("implement me")
+}
+
+type mockOSAnalyzer struct{}
+
+func (m mockOSAnalyzer) Analyze(extractor.FileMap) (OS, error) {
+	panic("implement me")
+}
+
+func (m mockOSAnalyzer) RequiredFiles() []string {
+	return []string{"file1", "file2", "file3"}
+}
+
+func TestAnalyze(t *testing.T) {
+	testCases := []struct {
+		name                string
+		saveLocalImageFunc  func(ctx context.Context, imageName string) (io.Reader, error)
+		extractFunc         func(ctx context.Context, imageName string, filenames []string) (extractor.FileMap, error)
+		extractFromFileFunc func(ctx context.Context, r io.Reader, filenames []string) (maps extractor.FileMap, e error)
+		expectedError       error
+		expectedFileMap     extractor.FileMap
+	}{
+		{
+			name: "happy path with docker installed and image found",
+			extractFromFileFunc: func(ctx context.Context, r io.Reader, filenames []string) (maps extractor.FileMap, e error) {
+				assert.Equal(t, []string{"file1", "file2", "file3"}, filenames)
```

Comment:
This assertion is checking the input arguments passing into `ExtractFromFile` as we want to make sure that the right filenames are passed in when operating.

Reply:
Yes. `AssertExpectations` can make sure that the function took the right arguments. IMHO, the assertion should be separated from test values in some way like the following. 
https://github.com/kubernetes/kubernetes/blob/4fda1207e347af92e649b59d60d48c7021ba0c54/pkg/kubelet/eviction/memory_threshold_notifier_test.go#L137

When a function is mixed in test values, it's difficult to read.

Reply:
When there are only values in test cases, I feel it's easy to read, but it may be personal preference.
https://github.com/kubernetes/kubernetes/blob/4fda1207e347af92e649b59d60d48c7021ba0c54/pkg/kubelet/eviction/memory_threshold_notifier_test.go#L115-L129

Reply:
After a second look I do agree its bit of an overkill to check input args in this case. Let's remove them for now and add them later through `AssertExpectations`